src/test/java/com/mongodb/jdbc/MongoDatabaseMetaDataTest.java

```diff
@@ -318,32 +252,92 @@ void testGetAttributes() throws SQLException {
                     "SCOPE_TABLE",
                     "SOURCE_DATA_TYPE",
                 };
-        for (int i = 0; i < columns.length; ++i) {
-            assertEquals(rsmd.getColumnName(i + 1), columns[i]);
-            assertEquals(rsmd.getColumnLabel(i + 1), columns[i]);
-        }
-        assertFalse(rs.next());
+
+        ResultSet rs = databaseMetaData.getAttributes(null, null, null, null);
```

Comment:
I'm sorry, I totally missed this last time, but null is not always accepted and not always a synonym for no filter.
Here for example, typeNamePattern and attributeNamePattern are not supposed to support Null. It's probably Ok to say that Null = %, especially since this are empty ResultSet. But be mindful that it is not following the spec and tools will use a pattern, not Null, so it might be best to test with a pattern at least for the ones that are not empty to make sure that we behave correctly.

Reply:
Will do. We'll need to be sure to keep this in mind for the integration test tickets as well, since those will all deal with non-empty result sets.